akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorSpec.scala

```diff
@@ -762,31 +762,30 @@ class EventSourcedBehaviorSpec extends ScalaTestWithActorTestKit(EventSourcedBeh
           ctx ⇒
             counterWithSnapshotAndRetentionProbe(ctx, pid, snapshotProbe.ref, retentionProbe.ref)
               .snapshotEvery(2)
-              .withRetention(
-                RetentionCriteria(snapshotEveryNEvents = 2, keepNSnapshots = 2, deleteEventsOnSnapshot = false))))
+              .withRetention(RetentionCriteria(snapshotEveryNEvents = 3, keepNSnapshots = 2))))
 
-      persistentActor ! IncrementWithPersistAll(3)
+      persistentActor ! IncrementWithPersistAll(10)
       persistentActor ! GetValue(replyProbe.ref)
-      replyProbe.expectMessage(State(3, Vector(0, 1, 2)))
+      replyProbe.expectMessage(State(10, (0 until 10).toVector))
       snapshotProbe.expectMessage(Try(Done))
       retentionProbe.expectMessageType[Success[DeleteSnapshotsCompleted]].value match {
         case DeleteSnapshotsCompleted(Criteria(SnapshotSelectionCriteria(maxSequenceNr, _, minSequenceNr, _))) =>
-          maxSequenceNr shouldEqual 2
-          minSequenceNr shouldEqual 0
+          maxSequenceNr shouldEqual 9
+          minSequenceNr shouldEqual 3
```

Comment:
```scala
.snapshotEvery(2)
.withRetention(RetentionCriteria(snapshotEveryNEvents = 3, keepNSnapshots = 2)
```
* Store 0-9
* Snapshot delete from 3 to 9
* Store 10-19
* Snapshot delete from 13 to 19